pkg/controller/podautoscaler/context/context.go

```diff
@@ -177,19 +182,20 @@ var annotationParsers = map[string]annotationParser{
 
 // UpdateByPaTypes should be invoked in any scaling context that embeds BaseScalingContext.
 func (b *baseScalingContext) UpdateByPaTypes(pa *autoscalingv1alpha1.PodAutoscaler) error {
-	source, err := autoscalingv1alpha1.GetPaMetricSources(*pa)
-	if err != nil {
-		return err
-	}
+	for _, ms := range pa.Spec.MetricsSources {
+		targetValue, err := strconv.ParseFloat(ms.TargetValue, 64)
+		if err != nil {
+			klog.ErrorS(err, "Failed to parse target value", "metric", ms.TargetMetric, "value", ms.TargetValue)
+			return fmt.Errorf("invalid targetValue for metric %q: %w", ms.TargetMetric, err)
```

Comment:
should we ignore the error and continue here? I am checking whether single failure will block the entire process. but parse should be super reliable